controllers/object_controls.go

```diff
@@ -1398,10 +1443,10 @@ func TransformDevicePlugin(obj *appsv1.DaemonSet, config *gpuv1.ClusterPolicySpe
 	if err != nil {
 		return err
 	}
-	obj.Spec.Template.Spec.Containers[0].Image = image
+	mainContainer.Image = image
```

Comment:
As a note: This is a simple but useful cleanup. The one comment is whether we want to always call this `mainContainer` or something more specific to make global searches simpler?

Reply:
Maybe `toolkitMainContainer` in the `TransformToolkit` function and `devicePluginMainContainer` in the `TransformDevicePlugin` function?